### DISCUSSION TITLE
Sonoff T2 (2 Gang) with a Sonoff D1 Dimmer -> How to make dimming work

### DISCUSSION THREAD

**Discussion by jackkitley** [arendst/Tasmota#19297] [Q&A]
Good morning

I have a Sonoff T2 (2 Gang) panel. My bottom button is used for the entrance light to our bedroom.  The top button i would like to make dimmable. 

Here is what i have done for the 2 gang switch:
![Screenshot 2023-08-11 at 08 39 35](https://github.com/arendst/Tasmota/assets/3243014/047c3fcd-766e-4191-827a-81ab754b2a56)

I have also set a device group for the dimmer and 2 gang. Turning the light on and off works well. The dimmer command on the 2 gang was unknown and hence why i added the PWM as a fake to enable the dimmer command hoping that it would relay the day via share to the dimmer but it doesnt work.

The PWM dimmer slider is attaching itself onto the bottom button for some reason...

stat/main_bedroom/RESULT = {"POWER2":"ON","Dimmer":15}

I need this Dimmer value to be sent on POWER1 so that it can communicate with the D1 dimmer but im not sure how to do this.

Here is the powering on of the devices:

05:54:19.477 MQT: stat/main_bedroom/RESULT = {"POWER1":"OFF"}
05:54:19.480 MQT: stat/main_bedroom/POWER1 = OFF (retained)

05:54:19.661 MQT: stat/bedroom_dimmer/RESULT = {"POWER":"OFF"}
05:54:19.664 MQT: stat/bedroom_dimmer/POWER = OFF

Any help would be appreciated.

Thanks

**Comment by sfromis** [comment]
In general, you should always start the gpio dropdown index at 1, and not skip any numbers for the same type. Tasmota does not implicitly pair gpios with the same index, but instead only based on order. Thus your config for gpio 5, 9 and 16 should be corrected. Only after that, you can start to see how it works.
